contribs/src/main/java/com/netflix/conductor/contribs/http/RestClientManager.java

```diff
@@ -30,9 +31,27 @@
 @Singleton
 public class RestClientManager {
 
-	private Client defaultClient = Client.create();
-	
-	public Client getClient(Input input) {
-		return defaultClient;
+	static final int DEFAULT_READ_TIMEOUT = 150;
+	static final int DEFAULT_CONNECT_TIMEOUT = 100;
+	static final String HTTP_TASK_READ_TIMEOUT = "http.task.read.timeout";
+	static final String HTTP_TASK_CONNECT_TIMEOUT = "http.task.connect.timeout";
+
+
+	private final ThreadLocal<Client> threadLocalClient;
+	private final int defaultReadTimeout;
+	private final int defaultConnectTimeout;
+
+	public RestClientManager(Configuration config) {
+		this.threadLocalClient = ThreadLocal.withInitial(()->Client.create());
+		this.defaultReadTimeout =config.getIntProperty(HTTP_TASK_READ_TIMEOUT, DEFAULT_READ_TIMEOUT);
+		this.defaultConnectTimeout = config.getIntProperty(HTTP_TASK_CONNECT_TIMEOUT, DEFAULT_CONNECT_TIMEOUT);
 	}
+
+	public Client getClient() {
```

Comment:
The Input parameter is actually useful in switching between the clients provided based on input. For eg: Returning client with vipAddress vs client with URL. We'll change this to interface to make it clear. But, please add it back for now to maintain backwards compatibility with existing implementations.

Reply:
I have added the input parameter back.